apps/site/pages/docs/api-extensions.mdx

```diff
@@ -317,30 +325,149 @@ export const fragment = gql`
           components, that can be used on multiple pages.
           <br />
         </td>
         <td>
-          General products data (StoreProduct) with the `totalCount` from
-          StorePageInfo.
+          General products data ([StoreProduct](https://v1.faststore.dev/reference/api/objects#storeproduct)) with the `totalCount` from [StorePageInfo](https://v1.faststore.dev/reference/api/objects#storepageinfo).
         </td>
         <td>
-          Frontend data from the `useLocalizedVariables()` and `useSession()`
-          hooks.
+          Frontend data from the `useLocalizedVariables()` and `useSession()` hooks.
         </td>
       </ExtendableQueryTable>
     </details>
   </li>
 </ul>
+
+---
+
+## How to consume FastStore API Extension
+
+
+Once you define which new fields will be exposed by the FastStore API Extension, then determine the GraphQL fragments where these fields will be used.
+Finally, it's time to know how to consume the new fields in your [New Sections](/docs/building-sections/creating-a-new-section) or [Custom Overridable Components](/docs/building-sections/overriding-components-and-props).
+
+It's important to know that FastStore exposes the data that comes from FastStore API along with FastStore API Extensions inside a provider called `PageProvider`.
+This data comes as a `context` in `PageProvider`, and you can use some hooks to access it:
+
+The `usePDP()` hook is recommendable in cases where you use your section on PDP.
+
+```ts copy
+import { usePDP } from "@faststore/core"
+
+const context = usePDP()
+
+```
+
+The `usePLP()` hook is recommendable in cases where you use your section on PLP.
+
+```ts copy
+import { usePLP } from "@faststore/core"
+
+const context = usePLP()
+
+```
+
+The `useSearchPage()` hook is recommendable in cases where you use your section on the Search Page.
+
+```ts copy
+import { useSearchPage } from "@faststore/core"
+
+const context = useSearchPage()
+
+```
+
+The `usePage()` hook is recommendable in cases where you have one section that is used in more than one type of page.
+
+```ts copy
+import { usePage } from "@faststore/core"
+
+const context = usePage()
+
+```
+
+This hook returns one of the following types as context: `PDPContext`, `PLPContext`, or `SearchPageContext`, and you can decide how to handle it depending on the page that will use this hook by passing the types as generics.
+
+```ts copy
+import { usePage } from "@faststore/core"
+
+const context = usePage<PLPContext | SearchPageContext>()
+
+```
+
+Also, you can use type assertion functions so that you can leverage the use of typescript and get the correct types
+
+```ts copy
+import { isPDP, isPLP, isSearchPage } from "@faststore/core";
+
+isPDP(context)
+isPLP(context)
+isSearchPage(context)
+
+```
+
+### Consume API Extensions data from new `Sections`
+
+Creates a new section using the [Creating a New Section](/docs/building-sections/creating-a-new-section) doc, and uses one of the hooks mentioned above.
```

Comment:
I'm not sure if I get here 🤔 Do I create the new section using the doc and then I use one of the hooks?

Reply:
Yeah, that's almost the same example from the other doc. But here we added the hook.